data_diff/diff_tables.py

```diff
@@ -322,45 +330,79 @@ def diff_tables(self, table1: TableSegment, table2: TableSegment) -> DiffResult:
         if self.bisection_factor < 2:
             raise ValueError("Must have at least two segments per iteration (i.e. bisection_factor >= 2)")
 
-        # Query and validate schema
-        table1, table2 = self._threaded_call("with_schema", [table1, table2])
-        self._validate_and_adjust_columns(table1, table2)
+        if is_tracking_enabled():
+            options = dict(self)
+            event_json = create_start_event_json(options)
+            send_event_json(event_json)
 
-        key_type = table1._schema[table1.key_column]
-        key_type2 = table2._schema[table2.key_column]
-        if not isinstance(key_type, IKey):
-            raise NotImplementedError(f"Cannot use column of type {key_type} as a key")
-        if not isinstance(key_type2, IKey):
-            raise NotImplementedError(f"Cannot use column of type {key_type2} as a key")
-        assert key_type.python_type is key_type2.python_type
-
-        # Query min/max values
-        key_ranges = self._threaded_call_as_completed("query_key_range", [table1, table2])
+        self.stats["diff_count"] = 0
+        start = time.time()
```

Comment:
**nit:** `time.time()` is not monotonic, it can go backward or jump forward, thus distorting the measurements — more likely on local time/clock adjustments by ntp or by the user.

For performance measurements, `time.monotonic()` or `time.perf_counter()` are used. For data-diff's scale, the first one is sufficient (the second one has higher precision, which is not needed).

However, in the majority of cases, `time.time()` will work sufficiently good too.